test/livebook/storage/ets_test.exs

```diff
@@ -87,4 +87,42 @@ defmodule Livebook.Storage.EtsTest do
       assert [] = Ets.all(:unknown_namespace)
     end
   end
+
+  describe "persistence" do
+    defp load_state(%{tmp_dir: tmp_dir, secret: secret, state: state}) do
+      %{state | secret: secret, file_path: Path.join([tmp_dir, "config.txt"])}
+    end
+
+    @table_name __MODULE__
+    setup do
+      # intentionally don't pass the 'file_path' nor 'secret' into init/1
+      # to avoid loading the file on initialization
+      {:ok, state} = Ets.init(table_name: @table_name)
+
+      [state: state, secret: :crypto.strong_rand_bytes(32)]
+    end
+
+    @tag :tmp_dir
+    test "saves/loads the ets content to/from an encrypted file", ctx do
+      state = load_state(ctx)
+
+      {:reply, :ok, state} = Ets.handle_call({:insert, :all, "test1", key: "val"}, {}, state)
+      {:reply, :ok, state} = Ets.handle_call({:insert, :all, "test2", key: "val"}, {}, state)
+
+      assert {:noreply, _state} = Ets.handle_info(:persist, state)
```

Comment:
These are internal details that we shouldn't be concerned with :)

One approach I can think of is creating one storage, adding entries and persisting it, then creating another from the same file. We could make the process name configurable to enable this, but the table name in the fetch callbacks is hardcoded. @josevalim any ideas?

Reply:
I know that those are internals but I couldn't find a better way to test it so that we don't interfere with the global Ets process gets started along with the application. In the end, if a person changes the internals then they automatically have to change the tests. In this case it is probably not that risky as no one will probably depend on it 🤷 

When it comes to making the process name configurable, the problem is that the behaviour callbacks don't expect another parameter which in this case could look like this `def fetch(process \\ __MODULE__, namespace, entity_id)` but I don't know if that is right for a contract to look like this.

Reply:
I think you can write, using the Livebook.Storage API, a special key, such as `:livebook_persistent_test`. Then in the test you load the ets table from the location and see if the key is there. This should be doable especially if we move to persisting on every write.

Reply:
@josevalim what do you mean by loading? We have only a single storage, and loading into that may override some other stuff, no?

Reply:
I mean the test can directly call `:ets.file2tab` to load it but then I remembered that the file also stores if the table is named and we currently use a named table for concurrent reads.

I guess one option would be to **not** name the table and instead store the table on `persistent_term`. Then we can read the table by doing `:persistent_term.get(Livebook.Storage.ETS)`. This way we can load multiple versions of the table (one in tests) and keep the other one in memory without one affecting the other.

WDYT?

Reply:
Ah, so you mean checking the key directly in the loaded table, not the storage on top of it. Ideally we would test loading part as well, but to do that we would need to instantiate the storage. Either way, not storing the table name in file sounds better!